addon/model.js

```diff
@@ -149,22 +150,23 @@ export default class MegamorphicModel extends EmberObject {
 
         // Update the mapping to point to the proxy instead of the instance
         recordDataToRecordMap.set(instance._recordData, proxy);
-
-        return proxy;
+        value = proxy;
       }
       if (DEBUG) {
         if (useNative === false) {
           let proxy = new Proxy(instance, megamorphicNativeDeprecationHandler);
 
           // Update the mapping to point to the proxy instead of the instance
           recordDataToRecordMap.set(instance._recordData, proxy);
-
-          return proxy;
+          value = proxy;
         }
       }
     }
-
-    return instance;
+    if (!value._topModel) {
+      value._topModel = value;
+    }
+    value._flushInitProperties();
```

Comment:
calling `_flushInitProperties` in `create` is sus.

Are we even still batching the initial properties?

I think it's possible this isn't needed with the versions of ember in support.

Reply:
hmm well I guess I'm wrong, we still need to batch, it's just that we now have a stable call order for the sets of versions we support.

Reply:
it used to be in the constructor, but we want to now call it after we have set the `_topModel` to avoid double computing values.